src/web/tokenserver.rs

```diff
@@ -138,128 +189,43 @@ pub fn get_sync(
-            }
-            Ok(x) => x.extract::<String>().unwrap(),
-        };
-
-        let fxa_kid = format!(
-            "{:013}-{:}",
-            user_record[0].keys_changed_at.unwrap_or(0),
-            client_state_b64
-        );
-        let thedict = [
-            ("node", &user_record[0].node),
-            ("fxa_kid", &fxa_kid), // userid component of authorization email
-            ("fxa_uid", &token_data.claims.sub),
-            ("hashed_device_id", &hashed_device_id),
-            ("hashed_fxa_uid", &hashed_fxa_uid),
-        ]
-        .into_py_dict(py);
-        // todo don't hardcode
-        // we're supposed to check the "duration" query
-        // param and use that if present (for testing)
-        thedict.set_item("expires", 300).unwrap(); // todo this needs to be converted to timestamp int (now + value * 1000)
-        thedict.set_item("uid", user_record[0].uid).unwrap();
-        let result = match tokenlib.call1("make_token", (thedict, &shared_secret)) {
-            Err(e) => {
-                e.print_and_set_sys_last_vars(py);
-                return Err(e);
-            }
-            Ok(x) => x.extract::<String>().unwrap(),
-        };
-        let derived_result = match tokenlib.call1("get_derived_secret", (&result, &shared_secret)) {
-            Err(e) => {
-                e.print_and_set_sys_last_vars(py);
-                return Err(e);
-            }
-            Ok(x) => x.extract::<String>().unwrap(),
-        };
-        //assert_eq!(result, false);
-        Ok((result, derived_result))
-    })
-    .unwrap();
+    let client_state_b64 =
+        base64::encode_config(&user_record[0].client_state, base64::URL_SAFE_NO_PAD);
+    let fxa_metrics_hash_secret = env::var("FXA_METRICS_HASH_SECRET")
+        .unwrap_or_else(|_| "insecure".to_string())
+        .into_bytes();
+    let hashed_fxa_uid = fxa_metrics_hash(&email, &fxa_metrics_hash_secret);
+    let device_id = "none".to_string();
+    let hashed_device_id = hash_device_id(&hashed_fxa_uid, &device_id, &fxa_metrics_hash_secret);
+
+    let fxa_kid = format!(
+        "{:013}-{:}",
+        user_record[0].keys_changed_at.unwrap_or(0),
+        client_state_b64
+    );
+    let (python_result, python_derived_result) =
+        Python::with_gil(|py| -> Result<(String, String), PyErr> {
+            let thedict = [
+                ("node", &user_record[0].node),
+                ("fxa_kid", &fxa_kid), // userid component of authorization email
+                ("fxa_uid", &token_data.claims.sub),
+                ("hashed_device_id", &hashed_device_id),
+                ("hashed_fxa_uid", &hashed_fxa_uid),
+            ]
+            .into_py_dict(py);
+            // todo don't hardcode
+            // we're supposed to check the "duration" query
+            // param and use that if present (for testing)
+            thedict.set_item("expires", 300).unwrap(); // todo this needs to be converted to timestamp int (now + value * 1000)
```

Comment:
You could have this pull from `env` like you're doing above with an `unwrap_or_else` that uses 300 as the default.

It'd also be nice to have a section of `Settings` that defined all of these so that there's no risk of these getting lost in the code.

Reply:
@jrconlin I think this was done by someone else; I think it ended up in the diff since I changed the indentation or something 🤔 Do you think it's worth making this change here? Might be worth waiting for when I do the work for #1052, but happy to make the change here if you'd prefer that!

Reply:
Well, not for this iteration. I was just noting the "todo". If it was something critical or important, I wouldn't have approved the changes 😉